src/lib.rs

```diff
@@ -550,6 +549,15 @@ where
     }
 }
 
+impl<M> Drop for Pool<M>
+where
+    M: ManageConnection,
+{
+    fn drop(&mut self) {
+        reap_connections(&self.0, true);
```

Comment:
Calling into reap_connections will immediately try to replace the dropped connections with new ones, and in any case, dropping the pool should immediately drop the list of connections in its internal state anyway.